### ISSUE TITLE
Fill in / impute short foreign domestic holes

### ISSUE THREAD

**Issue by korenmiklos** [korenmiklos/expat-analysis#38]

**Comment by korenmiklos** [comment]
```
replace foreign = 1 if l.foreign==1 & l2.foreign==1 & f.foreign==1 & f2.foreign==1
replace foreign = 0 if l.foreign==0 & l2.foreign==0 & f.foreign==0 & f2.foreign==0
bysort frame_id_numeric: egen minyear = min(year)
bysort frame_id_numeric: egen maxyear = max(year)
replace foreign = 1 if l.foreign==1 & f.foreign==1 & f2.foreign==1 & year==minyear+1
replace foreign = 1 if l.foreign==1 & l2.foreign==1 & f.foreign==1 & year==minyear+1
replace foreign = 0 if l.foreign==0 & f.foreign==0 & f2.foreign==0 & year==maxyear-1
replace foreign = 0 if l.foreign==0 & l2.foreign==0 & f.foreign==0 & year==maxyear-1
```

**Comment by zaveczgergo** [comment]
Doing this in the select_sample changes the number of foreign rows from 83.723 to 83.788 (out of 491.680). In other words, there is no huge change (it is not the final analysis_sample number).
I had to slightly modify the code as there is no frame_id_numeric at the point where the interpolation is asked for, so I had to move it later.
I have not commited select_sample yet, as I am waiting for information on the `foundyear` variable crash question ( issue #43).
